flake.nix

```diff
@@ -11,14 +11,19 @@
     lib = {
       hm = import ./modules/lib { lib = nixpkgs.lib; };
       homeManagerConfiguration = { configuration, system, homeDirectory
-        , username, extraSpecialArgs ? { }
+        , username, stateVersion ? "20.09", extraModules ? [ ]
+        , extraSpecialArgs ? { }
         , pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages
         , check ? true }@args:
+        assert nixpkgs.lib.versionAtLeast stateVersion "20.09";
+
         import ./modules {
           inherit pkgs check extraSpecialArgs;
+
           configuration = { ... }: {
-            imports = [ configuration ];
-            home = { inherit homeDirectory username; };
+            imports = [ configuration ] ++ extraModules;
```

Comment:
I'm not sure the additional cognitive load of `extraModules` is justified. I find NixOS modules automagic (in a negative way) enough, so I think we should stick to the pattern people will be more familiar about, namely adding modules in `imports`.

Reply:
If we wanted to mimick the API of NixOS' `nixosSystem`, we would replace the `extraModules` and `configuration` with `modules` but it is what it is now.

Reply:
Perhaps I should omit this if we want to migrate just to `modules` instead? The reason I included this was that I felt it was nice to separate user level config from library modules that doesn't actually apply any configuration, but just expose new options that the user can use.